network/tcp.go

```diff
@@ -379,38 +378,37 @@ func (t *TCPListener) Listening() bool {
 
 // TCPHost implements the Host interface using TCP connections.
 type TCPHost struct {
-	addr  Address
 	suite Suite
+	sid   *ServerIdentity
 	*TCPListener
 }
 
 // NewTCPHost returns a new Host using TCP connection based type.
-func NewTCPHost(addr Address, s Suite) (*TCPHost, error) {
+func NewTCPHost(sid *ServerIdentity, s Suite) (*TCPHost, error) {
```

Comment:
Why not making a `NewTLSHost` ?
It seems cleaner to separate the two, and maybe write a `deprecated plain TCP connection` on the log or stg ?

Reply:
Honestly, this is the first thing that worked, and I was too tired to change it more. Assuming TCP happened all over the tests.

Reply:
This is the only thing that I still _would_ still change - a `NewTLSHost` makes sense for me.

Reply:
I have looked into this some. If one really wants to "clean this up", and I don't see what it messy and needs cleaning, then you'd want to kill New*Host entirely and have only NewHost(...) (Host, error). I've started on that, but then tests do not compile because they are written assuming that they can muck around inside the returned *TCPHost.

TLS happens inside of TCP. I do not see the problem of having TCP with TLS and PlainTCP (i.e. TCP without TLS) sharing the same data structure.